main/src/ca/uwaterloo/flix/language/phase/Parser.scala

```diff
@@ -905,6 +905,16 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
       }
     }
 
+    def MatchType: Rule1[ParsedAst.Expression.MatchType] = {
+      def Rule: Rule1[ParsedAst.MatchTypeRule] = rule {
+          keyword("case") ~ WS ~ Names.Variable ~ optWS ~ ":" ~ optWS ~ Type ~ optWS ~ atomic("=>") ~ optWS ~ Stm ~> ParsedAst.MatchTypeRule
+      }
+
+      rule {
+        SP ~ keyword("matchType") ~ WS ~ Expression ~ optWS ~ "{" ~ optWS ~ oneOrMore(Rule).separatedBy(CaseSeparator) ~ optWS ~ "}" ~ SP ~> ParsedAst.Expression.MatchType
```

Comment:
I rather liked the keyword `typematch`. I don't think we should have camel case keywords.